typing/printtyp.ml

```diff
@@ -936,16 +936,16 @@ end = struct
             (* Some part of the type we've already printed has assigned another
              * unification variable to that name. We want to keep the name, so
              * try adding a number until we find a name that's not taken. *)
-            let current_name = ref name in
-            let i = ref 0 in
-            while List.exists
-                    (fun (_, name') -> !current_name = name')
-                    !names
-            do
-              current_name := name ^ (Int.to_string !i);
-              i := !i + 1;
-            done;
-            !current_name
+            let available name =
+              List.exists
```

Comment:
I guess we're still linear in the length of this list, but it doesn't matter in practice?

Reply:
I guess the real question is, how large does the `names` list get, and if it gets large, maybe we should change from a list to a different data structure.

Reply:
I consider this as orthogonal to the use of `find_first_mono` in this function, although you are right that we probably don't get a good worst-case complexity in this case. I'm okay with that -- I propose to introduce `find_first_mono` and use it everywhere applicable, even if it does not solve all complexity issues.